src/api/routes/auth.js

```diff
@@ -60,4 +62,39 @@ export default (app) => {
             return next(err);
         }
     });
+
+    router.post("/recover/request", validators.recover, async (req, res, next) => {
+        try {
+            const accountService = new AccountService();
+
+            const account = await accountService.findByEmail(req.body.email);
+
+            if (account === null) {
+                return res.status(HTTPStatus.OK).json({});
+            }
+
+            const link = accountService.buildPasswordRecoveryLink(account);
+            accountService.sendPasswordRecoveryNotification(account, link);
+
+            return res.status(HTTPStatus.OK).json({});
+        } catch (err) {
+            console.error(err);
+            return next(err);
+        }
+    });
+
+    router.get("/recover/:token/confirm", validators.confirmRecover, validToken, (req, res) => res.status(HTTPStatus.OK).json({}));
+
+    router.post("/recover/:token/confirm", validators.finishRecover, validToken, async (req, res, next) => {
+        const { email } = verifyAndDecodeToken(req.params.token, env.jwt_secret);
```

Comment:
We are validating and decoding the token twice, once in the validator and another time here. Maybe the validator can assign some information to `req.locals` like what sometimes happens for auth validators?

Reply:
I talked about this with @imnotteixeira a few weeks ago. It creates dependencies and since the function is not very heavy, we should probsbly not worry about this small optimization.

I think req.locals would be usefull in other routes that make heavier database operations though.